src/Maps/MapReader.cpp

```diff
@@ -128,7 +128,7 @@ namespace MapReader {
 		{
 			uint32_t numTileGroups;
 			streamReader.Read(numTileGroups);
-			streamReader.SeekRelative(4);
+			streamReader.SeekRelative(sizeof(uint32_t)); // Seek past unknown field
```

Comment:
This could become a read into a temporary variable, rather than a seek.

We should be consistent between reading and writing about `int32_t`/`uint32_t`.

I would be fine with having `MapData` track an unknown value. That seems like quite a reasonable way to handle it. I'd also be happy with dummy read/writes for that field. For new maps, you still need to consider how the value will be initialized, so you can't entirely get away from the initialization issue.

As for the game not using the value, that applies to all of the tile group data. The game stops reading before processing that entire section. I think it was intended for a map editor, which was never publicly released.